src/unix/solarish/compat.rs

```diff
@@ -169,3 +169,42 @@ pub unsafe fn forkpty(
 
     0
 }
+/// Standard conforming wrapper around `native_getpwent_r()`
+///
+/// TODO: remove this entry and rename `native_getpwent_r` to `getpwent_r` in
+/// `mod.rs` when the standard conforming definitoin is added to the upstream
+/// kernel
```

Comment:
I would probably remove this whole doc comment and the matching one on the next function, for a few reasons:

- I don't think we usually put doc comments on any public symbols in the libc crate?
- It won't actually be strictly necessary to do this, as the existing `getpwent_r()` will never go away or change behaviour (illumos does not break backwards compatibility for public symbols) and in fact it may be counter productive to do it too soon (for the same reason we're adding it in the first place)
- It is not actually a function that conforms to any standard of which I'm aware; rather it's a GNU extension with regrettably different arguments